[ews-build.webkit.org] Remove Python 2 testing

JonWBedard opened the pull request:
#### ca0cd2ce84fe5a68949eb066c8412bca969aabda
<pre>
[ews-build.webkit.org] Remove Python 2 testing
<a href="https://rdar.apple.com/128020540">rdar://128020540</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=274111">https://bugs.webkit.org/show_bug.cgi?id=274111</a>

Reviewed by Aakash Jain and Alexey Proskuryakov.

* Tools/CISupport/ews-build/events.py:
(Events): Collapse Python 2 and Python 3 testing events into a single action.
* Tools/CISupport/ews-build/factories.py:
(WebKitPyFactory.__init__): Collapse Python 2 and Python 3 testing.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps): Collapse Python 2 and Python 3 testing.
* Tools/CISupport/ews-build/steps.py:
(RunWebKitPyTests): Renamed from WebKitPyTest.
(RunWebKitPyTests.getResultSummary):
(WebKitPyTest): Renamed to RunWebKitPyTests.
(RunWebKitPyPython2Tests): Deleted.
(RunWebKitPyPython3Tests): Moved to RunWebKitPyTests.
* Tools/CISupport/ews-build/steps_unittest.py:
(test_success):
(test_unexpected_failure):

Canonical link: <a href="https://commits.webkit.org/278719@main">https://commits.webkit.org/278719@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/82d76389df7aea93281ddbde93b00397e450f32c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/51406 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30716 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3757 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/54673 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/2099 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/53709 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/37034 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1779 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/54673 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53505 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/28361 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/44332 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/54673 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/51255 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/25671 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/1591 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/47638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1668 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/56265 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/50910 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26525 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/1553 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/49259 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/51425 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/27765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/44392 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/48440 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/28658 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7482 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/27500 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
rs=me

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3073,15 +3073,6 @@ def getResultSummary(self):
-    command = ['python', 'Tools/Scripts/test-webkitpy', '--verbose', '--json-output={0}'.format(jsonFileName)]
-
-
 class RunWebKitPyPython3Tests(WebKitPyTest):
```
We can consider removing this class as well, and just modify base class appropriately.

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3073,15 +3073,6 @@ def getResultSummary(self):
-
 class RunWebKitPyPython3Tests(WebKitPyTest):
     language = 'python3'
     name = 'webkitpy-tests-{}'.format(language)
```
now that there is no more python2 in this code, we can consider removing explicit python3 from these step names.

JonWBedard replied:
Good point, let me do that...

JonWBedard committed:
[ews-build.webkit.org] Remove Python 2 testing
rdar://128020540
https://bugs.webkit.org/show_bug.cgi?id=274111

Reviewed by Aakash Jain and Alexey Proskuryakov.

* Tools/CISupport/ews-build/events.py:
(Events): Collapse Python 2 and Python 3 testing events into a single action.
* Tools/CISupport/ews-build/factories.py:
(WebKitPyFactory.__init__): Collapse Python 2 and Python 3 testing.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps): Collapse Python 2 and Python 3 testing.
* Tools/CISupport/ews-build/steps.py:
(RunWebKitPyTests): Renamed from WebKitPyTest.
(RunWebKitPyTests.getResultSummary):
(WebKitPyTest): Renamed to RunWebKitPyTests.
(RunWebKitPyPython2Tests): Deleted.
(RunWebKitPyPython3Tests): Moved to RunWebKitPyTests.
* Tools/CISupport/ews-build/steps_unittest.py:
(test_success):
(test_unexpected_failure):

Canonical link: https://commits.webkit.org/278719@main
